script/utils/update-redirects-data.ts

```diff
@@ -60,10 +75,20 @@ const updateRedirectsData = async(currentBook: BookWithOSWebData, newBook: BookW
       countNewRedirections++;
     // remove `else` to enable legitimately removing pages from books
     // only once uuids are guaranteed to be consistent
-    } else if (!newSection && matchSlug(section.slug) === undefined && !matchException) {
+    } else if (!newSection && matchSlug(section.slug) === undefined && !matchException && !booksAreDifferent) {
       throw new Error(
         `updateRedirects prohibits removing pages from a book, `
         + `but neither section with ID ${section.id} nor slug ${section.slug} was found in book ${newBook.id}`);
+    } else if (booksAreDifferent && !redirects.find(matchRedirect(section))) {
+      // if redirecting a book but no page match found, check for a canonical page
+      const canonicalPageMap = CANONICAL_MAP[currentBook.id]?.find((pageMap) => pageMap[0] === newBook.id) || [];
+      const canonicalPageId = canonicalPageMap[1] && canonicalPageMap[1][section.id];
+      // if no canonical page, fall back to default book page
+      const redirectSection = (canonicalPageId && findArchiveTreeNodeById(newBook.tree, canonicalPageId))
```

Comment:
lets also add a code comment in here that this condition case is specific to the retirement logic